lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -345,22 +345,26 @@ export default class DocumentComposer {
       }
 
       // `serviceEndpoint` validations.
-      const serviceEndpoint = service.serviceEndpoint;
-      if (typeof serviceEndpoint === 'string') {
-        const uri = URI.parse(service.serviceEndpoint);
-        if (uri.error !== undefined) {
-          throw new SidetreeError(
-            ErrorCode.DocumentComposerPatchServiceEndpointStringNotValidUri,
-            `Service endpoint string '${serviceEndpoint}' is not a valid URI.`
-          );
-        }
-      } else if (typeof serviceEndpoint === 'object') {
-        // Allow `object` type only if it is not an array.
-        if (Array.isArray(serviceEndpoint)) {
-          throw new SidetreeError(ErrorCode.DocumentComposerPatchServiceEndpointCannotBeAnArray);
+      // transform URI strings and JSON objects into array so that we can run validations more easily
```

Comment:
Thanks for the awesome change! Can you copy the change under `versions/latest` (so far just this file) and overwrite [versions/1.0](https://github.com/decentralized-identity/sidetree/tree/master/lib/core/versions/1.0) folder also?  I don't think the change is worth divergent behavior and cutting a v1.1 for.